orca/request_opendap.py

```diff
@@ -1,21 +1,30 @@
 import re
-from xarray import open_dataset
+from xarray import open_dataset, load_dataset
+from tempfile import NamedTemporaryFile
 
+from .split import split_url
 
-def request_opendap(url, out_file):
-    try:
-        d = open_dataset(url)
-        d.to_netcdf(out_file)
-    except Exception as e:
-        raise e
 
-    return out_file
+def request_opendap(url):
+    data = open_dataset(url)
+    size = data.nbytes
+
+    if size / 2 > 5e8:
+        urls = split_url(url, size)
+    else:
+        urls = [url]
+
+    output = []
+    for path in urls:
+        tmp = NamedTemporaryFile(suffix=".nc", dir="/tmp")
+        open_dataset(path).to_netcdf(tmp.name)
+        output.append(tmp)
+
+    return output
 
 
 def build_url(thredds_base, filepath, variable, lat, lon):
     variable_form = re.compile(r"([a-z]*)(\[\d*(:\d*){1,2}\])")
     var_name, var_time = variable_form.findall(variable)[0][:2]
 
-    # Expected format
-    # ?lon[0:1:1067],lat[0:1:509],time[0:1:55114],tasmax[0:1:0][0:1:0][0:1:0]
-    return f"{thredds_base}{filepath}?lon{lon},lat{lat},time{var_time},{var_name}[0:1:0][0:1:0][0:1:0]"
+    return f"{thredds_base}{filepath}?{var_name}{var_time}{lat}{lon}"
```

Comment:
This is a little different than the format we were using before. How does this behave compared to what we had before?

Reply:
Sorry, I thought we agreed on changing it to this for some reason. I think I don't fully understand the netcdf split to be honest but I found that splitting on `{var_name}{var_time}{lat}{lon}` eg `tasmax[0:1:0][0:1:509][0:1:1067]` was the easiest way/ only way I could figure out how to significantly change the file size

Reply:
That works just fine. I just forgot about that over the weekend :+1: